hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/grid/AbstractGrid.java

```diff
@@ -8,40 +8,70 @@
  */
 package org.eclipse.hawkbit.ui.common.grid;
 
+import java.util.Arrays;
+import java.util.Locale;
+
 import org.eclipse.hawkbit.ui.SpPermissionChecker;
 import org.eclipse.hawkbit.ui.components.RefreshableContainer;
-import org.eclipse.hawkbit.ui.utils.VaadinMessageSource;
+import org.eclipse.hawkbit.ui.utils.SPDateTimeUtil;
 import org.eclipse.hawkbit.ui.utils.SPUIDefinitions;
+import org.eclipse.hawkbit.ui.utils.VaadinMessageSource;
 import org.vaadin.addons.lazyquerycontainer.LazyQueryContainer;
 import org.vaadin.spring.events.EventBus;
 import org.vaadin.spring.events.EventBus.UIEventBus;
 
-import com.vaadin.data.Container;
 import com.vaadin.data.Container.Indexed;
+import com.vaadin.data.util.GeneratedPropertyContainer;
+import com.vaadin.data.util.converter.Converter;
 import com.vaadin.ui.Grid;
 
 /**
- * Abstract table class.
+ * Abstract grid that offers various capabilities (aka support) to offer
+ * convenient enhancements to the vaadin standard grid.
  *
+ * @param <T>
+ *            The container-type used by the grid
  */
-public abstract class AbstractGrid extends Grid implements RefreshableContainer {
-
+public abstract class AbstractGrid<T extends Indexed> extends Grid implements RefreshableContainer {
```

Comment:
A name of generic type should not have more than one character, because you cannot distinguish it from a java-class.

See sonar rule S00119. and java naming convention
http://docs.oracle.com/javase/tutorial/java/generics/types.html